### PR TITLE
add pl_glue example test

### PR DESCRIPTION
**Pull request by stas00**
Currently PL glue example isn't being tested by CI. This PR fixes that.
https://github.com/huggingface/transformers/pull/6027

the added test currently fails (goodness) and fixes are being PR'ed in 
1. https://github.com/huggingface/transformers/pull/6027
2. https://github.com/huggingface/transformers/pull/6028

At the moment this is a basic 'can-run' test - I will add an actual validation once the `pl_glue_run.py` script is made to work.

### CODE REVIEW

**Review by sshleifer — APPROVED** [review]
Thanks!

**Comment by stas00** [comment]
Looks like this PR needs one of these merged first: https://github.com/huggingface/transformers/pull/6028 or https://github.com/huggingface/transformers/pull/6027, as `-n_gpus` is now required in `lightning_base.py`

**Comment by stas00** [comment]
No matter what I try CI can't get to acc/f1>=0.75, even though I get 1.0 on my machine. Suggestions?

**Comment by stas00** [comment]
OK, as suggested by @sshleifer I changed the PR to only test `run_pl_glue.py` is able to run and complete its job. Will have to figure out quality another time.

**Comment by stas00** [comment]
CI failure is totally unrelated.

File: examples/test_examples.py [hunk 1]
```diff
@@ -76,6 +79,41 @@ def test_run_glue(self):
...
+            --max_seq_length=128
+            """.split()
+
+        if torch.cuda.is_available():
+            testargs += ["--fp16", "--gpus=1"]
+
+        with patch.object(sys, "argv", testargs):
+            result = run_pl_glue.main()
+            # for now just testing that the script can run to a completion
+            self.assertGreater(result["acc"], 0.25)
```
Comment (LysandreJik) [inline examples/test_examples.py:107]: Let's not forget to update this later, 0.25 accuracy is not impressive for a 2-label dataset.

**Review by LysandreJik — APPROVED** [review]
Works for me!

**Comment by stas00** [comment]
I'm getting acc/f1=1 on my machine. I just don't know why CI gets a much worser score with the same config. I'm all ears to what else I can try, as you can see from the commits I tried a bunch of hparams tweaking to no avail. 

FWIW, @sshleifer was able to reproduce the low score on his machine.

File: examples/test_examples.py (hunk 1 above)
Comment (stas00) [reply examples/test_examples.py:107]: I'm getting acc/f1=1 on my machine. I just don't know why CI gets a much worser score with the same config. I'm all ears to what else I can try, as you can see from the commits I tried a bunch of hparams tweaking to no avail.

FWIW, @sshleifer was able to reproduce the low score on his machine.

**Pull request merged**